### PR TITLE
[4.0] Module cog

### PR DESCRIPTION
**Pull request by brianteeman**
All the modules in the admin control panel have a cog icon which lets you edit the module. Howeveer the permission check is not correct as you need to have access to the module component AND to edit the module.

### To test
Create a user with default manager permissions
Create a user with default administrator permissions
Change the permissions on one module so that only Super Users can edit the module

Login as manager - the cog is displayed on all the modules but you get a 403
Login as administrator - the cog is displayed and works on all the modules except the one you dont have permission to edit where you get a 403
Login as super user - the cog is displayed and works on all the modules

#### Apply the pr
Login as manager - the cog is not displayed at all
Login as administrator - the cog is displayed and works on all the modules except the one you dont have permission to edit where the cog is not displayed at all
Login as super user - the cog is displayed and works on all the modules

### CODE REVIEW

File: administrator/templates/atum/html/modules.php [hunk 1]
```diff
@@ -49,7 +49,7 @@ function modChrome_well($module, &$params, &$attribs)
 	{
 		// Permission checks
 		$user           = JFactory::getUser();
-		$canEdit	    = $user->authorise('core.edit', 'com_modules.module.' . $module->id);
+		$canEdit	    = $user->authorise('core.edit', 'com_modules.module.' . $module->id) || $user->authorise('core.modules.manage', 'com_modules');
```
Comment (Quy) [inline administrator/templates/atum/html/modules.php]: Align `=`

File: administrator/templates/atum/html/modules.php [hunk 2]
```diff
@@ -49,7 +49,7 @@ function modChrome_well($module, &$params, &$attribs)
 	{
 		// Permission checks
 		$user           = JFactory::getUser();
-		$canEdit	    = $user->authorise('core.edit', 'com_modules.module.' . $module->id);
+		$canEdit        = $user->authorise('core.edit', 'com_modules.module.' . $module->id) || $user->authorise('core.modules.manage', 'com_modules');
```
Comment (Quy) [inline administrator/templates/atum/html/modules.php:52]: > Howeveer the permission check is not correct as you need to have access to the module component AND to edit the module.

`||` should be `&&`

Comment (brianteeman) [reply administrator/templates/atum/html/modules.php:52]: that is exactly what I thought as well but when I tested it I did not get thee results I expected from the test scenarios. Could you double check please. Maybe I did something stupid

Comment (SharkyKZ) [reply administrator/templates/atum/html/modules.php:52]: I think it should be `core.manage`, not `core.modules.manage`.

**Comment by brianteeman** [comment]
Closed see #21932

**Pull request closed without merging**